### ISSUE TITLE
Compile error with fullTemplateTypeCheck: true

### ISSUE THREAD

**Issue by achimha** [swimlane/ngx-datatable#1240]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => search github for a similar issue or PR before submitting
```

**Current behavior**
Angular 5.2 introduces a new AoT compiler option to perform a more thorough type check on templates. It is expected that this option will default to true in a future version:

https://blog.angular.io/angular-5-2-now-available-312d1099bd81

Building a project with ngx-datatable shows two type errors:

`node_modules/@swimlane/ngx-datatable/release/components/footer/footer.component.d.ts.DataTableFooterComponent.html(8,9): : Property 'template' does not exist on type 'TemplateRef<any>'.
node_modules/@swimlane/ngx-datatable/release/components/body/body-row-wrapper.component.d.ts.DataTableRowWrapperComponent.html(5,7): : Expected 1 arguments, but got 0.`

* **Table version:** 11.1.7

* **Angular version:** 5.2.0

* **Browser:** [all ]

* **Language:** [TypeScript 2.5.2]

**Comment by petivagyok16** [comment]
Same here #1296

**Comment by atparkweb** [comment]
Same here and issue still exists in v11.2.0

**Comment by fgustovo** [comment]
same here v11.2.0 and angular 5.2.9

**Comment by agilob** [comment]
Fixed for me in 14.0.0
